pkgs/development/python-modules/torchbench/default.nix

```
@@ -0,0 +1,76 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  accelerate,
+  boto3,
+  beautifulsoup4,
+  distro,
+  iopath,
+  monkeytype,
+  numpy,
+  opencv4,
+  patch,
+  psutil,
+  py-cpuinfo,
+  pynvml,
+  pytest,
+  pytest-benchmark,
+  pyyaml,
+  requests,
+  submitit,
+  tabulate,
+  transformers,
+}:
+
+buildPythonPackage {
+  pname = "torchbench";
+  version = "0-unstable-2024-10-17";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "pytorch";
+    repo = "benchmark";
+    rev = "00c9b9ed61b70425e6875e0c4d6c5ad18a1b1d4c";
+    hash = "sha256-Sc50BvIQWVZgotOOC1sG3qjVHRbqxDcLH4xA+Vt/T3o=";
+    fetchSubmodules = true;
+  };
+
```

```suggestion
  postPatch = ''
    substituteInPlace torchbenchmark/__init__.py \
      --replace-fail "REPO_PATH = Path(os.path.abspath(__file__)).parent.parent" "REPO_PATH = Path('$src')"
  '';
```
A relative path outside the package called `REPO_PATH` is used in many places. We could replace all REPO_PATHs in this way.
I'm not sure that's the best approach.

https://github.com/search?q=repo%3Apytorch%2Fbenchmark%20REPO_PATH&type=code